src/renderer/components/contact/ContactList.tsx

```diff
@@ -6,6 +6,13 @@ import { debounceWithInit } from '../chat/ChatListHelpers'
 import { BackendRemote, Type } from '../../backend-com'
 import { selectedAccountId } from '../../ScreenController'
 
+/**
+ * Make sure not to render all of the user's contacts with this component.
+ * Some users might have thousands of contacts, which can result in
+ * this component taking seconds to render.
+ * Instead consider using 'react-window', like we do in the `CreateChat`
+ * component.
+ */
```

Comment:
I think it would make sense to have the list of Create chat also as component.

In the long term it would also make sense to move this `ContactList` component to use react-window and lazy loading, though it might be a bit more involved because of the extra features.

maybe we should rename `ContactList` to sth like `ContactListForEdit` or some better name that reflects the extra features when we have a dedicated virtual contact list.

Though maybe it's too much work for now to generalise the virtual list from CreateChat to it's own component because there is the question of how to best represent the pseudo/fake contact items in the api of such an component.

Reply:
Maybe we could have some contact list component with some api like

```tsx
interface PseudoItem {
   icon: IconName,
   label: string,
   clickAction: ()=>void,
   contextMenu?: ContextMenuItem
}

return <ContactList
   extraItemsInBeginning={PseudoItem[]}
   contactIds={number[]}
>
   { (contact: Contact) => <Contact contact={contact }/> }
   // so you could replace the contact item with alternative versions
   // that support either checkboxes or removal buttons:
   { 
      (contact: Contact) =>
      <RemovableContact
         contact={contact} 
         onRemove= {(contactId:number)=>{/* removal logic */}} />
   }
</ContactList>

```

I think sth like that could be an api improvement. But definitely out of scope for this pr.

Reply:
Yeah, I thought it's too early to make an abstraction just for these two places, especially considering that neither 'react-window' nor `ContactList` are too thick to inline.

Reply:
well technically there at least 3 cases/variants:

- normal: new chat dialog
- checkbox: add members to group
- remove button: group members

I think the contact list with it's many props is kinda hard to read code, like it could be simpler with such an abstraction.
react-window + infini-loader add quite some code we shouldn't copy paste into every component. anyways this is still general thoughts nothing for this pr.